Mathlib/RingTheory/IsTensorProduct.lean

```diff
@@ -471,4 +494,33 @@ theorem Algebra.IsPushout.algHom_ext [H : Algebra.IsPushout R S R' S'] {A : Type
   · intro s₁ s₂ e₁ e₂
     rw [map_add, map_add, e₁, e₂]
 
+/--
+Given a commutative diagram of rings
+```
+  R  →  S  →  T
+  ↓     ↓     ↓
+  R' →  S' →  T'
+```
+where the left-hand square is a pushout and the big rectangle is a pushout, then also the
+right-hand square is a pushout. Note that this is essentially the isomorphism
+`T ⊗[S] (S ⊗[R] R') ≃ₐ[T] T ⊗[R] R'`.
+-/
+lemma Algebra.IsPushout.of_comp {T' : Type*} [CommRing T'] [Algebra R T']
+    [Algebra S' T'] [Algebra S T'] [Algebra T T'] [Algebra R' T']
+    [IsScalarTower R T T'] [IsScalarTower S T T'] [IsScalarTower S S' T']
+    [IsScalarTower R R' T'] [IsScalarTower R S' T'] [IsScalarTower R' S' T']
+    [Algebra.IsPushout R S R' S'] [Algebra.IsPushout R T R' T'] :
+    Algebra.IsPushout S T S' T' := by
```

Comment:
Can you please also add an iff version?

Reply:
I replaced it with the `iff` version as this results in less code duplication. Should I provide the two directions as `comp` and `of_comp` separately or is this sufficient? (It certainly is for my application).

Reply:
This is perfect